spec/Factory/RefundPaymentFactorySpec.php

```diff
@@ -39,29 +38,45 @@ public function it_implements_refund_payment_factory_interface(): void
         $this->shouldImplement(RefundPaymentFactoryInterface::class);
     }
 
-    public function it_creates_new_refund_payment(
-        PaymentMethodRepositoryInterface $paymentMethodRepository,
+    public function it_creates_a_new_refund_payment(
+        OrderInterface $order,
         PaymentMethodInterface $paymentMethod,
-        OrderInterface $order
+        RefundPaymentInterface $refundPayment
     ): void {
-        $paymentMethodRepository->find(1)->willReturn($paymentMethod);
+        $refundPayment->getOrder()->willReturn($order);
+        $refundPayment->getAmount()->willReturn(1000);
+        $refundPayment->getCurrencyCode()->willReturn('USD');
+        $refundPayment->getState()->willReturn(RefundPaymentInterface::STATE_NEW);
+        $refundPayment->getPaymentMethod()->willReturn($paymentMethod);
 
         $this
             ->createWithData(
                 $order,
                 1000,
                 'USD',
                 RefundPaymentInterface::STATE_NEW,
-                1
+                $paymentMethod
             )
-            ->shouldBeLike(
-                new RefundPayment(
-                    $order->getWrappedObject(),
-                    1000,
-                    'USD',
-                    RefundPaymentInterface::STATE_NEW,
-                    $paymentMethod->getWrappedObject()
-                )
-            );
+            ->shouldBeSameAs($refundPayment)
```

Comment:
What was wrong with `shouldBeLike`? With this approach, we need to mock the expected result 🚀

Reply:
To be honest, I've used the exisiting comparator